### PR TITLE
Merge Feature/utilities into Master

### PR DESCRIPTION
**Pull request by sushma-4**
This branch contains refactored versions of all files in Utilities directory

### CODE REVIEW

**Comment by sushma-4** [comment]
@joshfactorial: Needs more testing for `gen_mut_model.py` , since no matching chromosomes were found in test files program not fully tested.

File: utilities/compute_gc.py [hunk 1]
```diff
@@ -113,25 +112,38 @@ def calculate_coverage(bin_dict: dict, window: int) -> float:
     return all_mean / float(running_total)
```
Comment (sushma-4) [inline utilities/compute_gc.py:112]: We still need to handle Divide_by_Zero error in some cases here @joshfactorial

Comment (joshfactorial) [reply utilities/compute_gc.py:112]: Changed the end of calculate_coverage to:

"""
    if running_total:
        return all_mean / float(running_total)
    else:
        return 0.0
"""

**Review by joshfactorial — COMMENTED** [review]
Excellent work. I'll run some further tests, but it all looks solid.

**Pull request merged**